master/buildbot/worker/ec2.py

```diff
@@ -539,24 +539,32 @@ def _wait_for_instance(self):
         else:
             self.failed_to_start(self.instance.id, self.instance.state['Name'])
 
-    def _wait_for_request(self, reservation):
+    def _thd_wait_for_request(self, reservation):
         duration = 0
         interval = self._poll_resolution
-        requests = self.ec2.meta.client.describe_spot_instance_requests(
-            SpotInstanceRequestIds=[reservation['SpotInstanceRequestId']])
-        request = requests['SpotInstanceRequests'][0]
-        request_status = request['Status']['Code']
-        while request_status in SPOT_REQUEST_PENDING_STATES:
+        while True:
+            # Sometimes it can take a second or so for the spot request to be
+            # ready.  If it isn't ready, you will get a "Spot instance reques
+            # ID 'sir-abcd1234' does not exist" exception.
+            try:
+                requests = self.ec2.meta.client.describe_spot_instance_requests(
+                    SpotInstanceRequestIds=[reservation['SpotInstanceRequestId']])
+            except:
```

Comment:
please specify the exception type

Reply:
Will do

Reply:
ping?

Reply:
Sorry, I was out last week.  I'll include that change when I add the test cases.

Reply:
Added exception type in my latest push